### PR TITLE
CB-13055 Fold all fetch options to `true`

### PR DESCRIPTION
**Pull request by raphinesse**
Manual constant folding assuming any `fetch` options to be `true`.

Extends #617

### CODE REVIEW

**Comment by raphinesse** [comment]
Will check out the failures on Travis :unamused:

**Comment by raphinesse** [comment]
Fixed `integration-tests/plugman_uninstall.spec.js`. Should be good now.

**Comment by raphinesse** [comment]
The test file that causes the errors on Windows is nasty. I'll DRY it first.

**Comment by raphinesse** [comment]
OK, I fixed up a few test corner cases in the first commit and added a commit that removes all usage of fetch options in the tests. Could you please take another look @dpogue?

**Comment by raphinesse** [comment]
OK, I'm done waiting for those awfully slow AppVeyor Tests just to see that one of them timed out again. :rage4:

File: src/plugman/fetch.js [hunk 1]
```diff
@@ -63,22 +63,10 @@ function fetchPlugin (plugin_src, plugins_dir, options) {
...
-            }
-
-            // Recurse and exit with the new options and truncated URL.
-            var new_dir = plugin_src.substring(0, plugin_src.indexOf('#'));
-
-            // skip the return if user asked for --fetch
-            // cordova-fetch doesn't need to strip out git-ref
-            if (!options.fetch) {
-                return fetchPlugin(new_dir, plugins_dir, options);
+                return Q.reject(new CordovaError('Cordova does not support subdirectories'));
```
Comment (janpio) [inline src/plugman/fetch.js]: Ugh, this broke the documented way to install plugin tests: https://github.com/apache/cordova-plugin-test-framework#tldr-try-it

Comment (janpio) [reply src/plugman/fetch.js]: This is also still documented at https://github.com/apache/cordova-cli/blob/master/doc/readme.md#plugin-spec

Comment (dpogue) [reply src/plugman/fetch.js]: That `:subdirectory` syntax was specific to the old Cordova fetching code (which has been removed), and is not supported by npm

Comment (janpio) [reply src/plugman/fetch.js]: https://github.com/apache/cordova-lib/issues/722
